src/mgmt/srv.c

```diff
@@ -92,12 +95,206 @@ process_killrank_request(Drpc__Call *drpc_req, Mgmt__DaosResponse *daos_resp)
 	mgmt__daos_rank__free_unpacked(pb_rank, NULL);
 }
 
+static void
+process_setrank_request(Drpc__Call *drpc_req, Mgmt__DaosResponse *daos_resp)
+{
+	Mgmt__SetRankReq	*pb_req = NULL;
+	int			rc;
+
+	mgmt__daos_response__init(daos_resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_req = mgmt__set_rank_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_req == NULL) {
+		daos_resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
+		D_ERROR("Failed to extract request\n");
+
+		return;
+	}
+
+	/* response status is populated with SUCCESS on init */
```

Comment:
is this comment in the right place? looks like it refers to response init

Reply:
This was copied from process_killrank_request, which I used as a template.

Reply:
I should fix that then :-)